bib/cmd/bootc-image-builder/image.go

```diff
@@ -126,7 +126,8 @@ func manifestForDiskImage(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest
 		return nil, fmt.Errorf("pipelines: no partition tables defined for %s", c.Architecture)
 	}
 	// XXX: extract into helper
-	if c.RootFSType != "" {
+	partitioningMode := disk.RawPartitioningMode
+	if c.RootFSType != "" && c.RootFSType != "btrfs" {
```

Comment:
Maybe via `switch/case` this could be slightly less nested, something like:
```go
diff --git a/bib/cmd/bootc-image-builder/image.go b/bib/cmd/bootc-image-builder/image.go
index 7017189..6e560bc 100644
--- a/bib/cmd/bootc-image-builder/image.go
+++ b/bib/cmd/bootc-image-builder/image.go
@@ -127,7 +127,10 @@ func manifestForDiskImage(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest
        }
        // XXX: extract into helper
        partitioningMode := disk.RawPartitioningMode
-       if c.RootFSType != "" && c.RootFSType != "btrfs" {
+       switch {
+       case c.RootFSType == "btrfs":
+               partitioningMode = disk.BtfrsPartitioningMode
+       case c.RootFSType != "":
                rootMountable := basept.FindMountable("/")
                if rootMountable == nil {
                        return nil, fmt.Errorf("no root filesystem defined in the partition table")
@@ -137,8 +140,6 @@ func manifestForDiskImage(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest
                        return nil, fmt.Errorf("root mountable is not an ordinary filesystem (btrfs is not yet supported)")
                }
                rootFS.Type = c.RootFSType
-       } else if c.RootFSType == "btrfs" {
-               partitioningMode = disk.BtfrsPartitioningMode
        }
 
        pt, err := disk.NewPartitionTable(&basept, c.Filesystems, DEFAULT_SIZE, partitioningMode, nil, rng)
```

Reply:
Should we merge and refactor this later?

Reply:
Let's refactor later.